### ISSUE TITLE
actions check error when installing dependency whose repo name != package name

### ISSUE THREAD

**Issue by jefferis** [r-lib/pak#521]
@gaborcsardi as discussed. Thanks so much for any input you can give.

tl;dr

Where in a package being checked with GitHub actions can I hint to pak about a dependency whose repo name != package name? So far, my best guess leaves a check error `It looks like this package has a loading problem when not on .libPaths:`. 

The details:

I am trying to use github actions with a package ([natverse/malevc](https://github.com/natverse/malevnc)). It depends (suggests) a package [natverse/fafbseg](https://github.com/natverse/fafbseg) which in turn depends (imports) a package [catmaid=natverse/rcatmaid](https://github.com/natverse/rcatmaid). For this last case the package name (**catmaid**) is not the same as the repo name (**rcatmaid**). This is not something I can change.

I know that I can do

```
pak::pkg_install('catmaid=natverse/rcatmaid')
```

But I do not know where I can hint to pak running through github actions that the package catmaid can be found at natverse/rcatmaid. The DESCRIPTION of the second package (fafbseg) has 

```
Remotes: 
    ...
    natverse/rcatmaid,
    ...
```

But this is not enough for pak (although it is sufficient for remotes) and I get a [dependency resolution error](https://github.com/natverse/malevnc/actions/runs/5456689261/jobs/9929842901#step:4:452). I have tried specifying the difficult dependency (catmaid) in the [workflow file](https://github.com/natverse/malevnc/blob/754556a042fda0cef11a9a958f6741b16c63fb17/.github/workflows/R-CMD-check.yaml) like this:

```yaml
      - uses: r-lib/actions/setup-r-dependencies@v2
        with:
          extra-packages: any::rcmdcheck, catmaid=natverse/rcatmaid
          needs: check
```
This installs fine, but then I get a [check error](https://github.com/natverse/malevnc/actions/runs/5462568425/jobs/9942102890#step:5:196):

```
* checking loading without being on the library search path ... WARNING
Error: Error in loadNamespace(x) : there is no package called ‘fafbseg’

It looks like this package has a loading problem when not on .libPaths:
see the messages for details.
```

**Comment by jefferis** [comment]
OK my conclusion is that this was the right thing to do:

```yml
      - uses: r-lib/actions/setup-r-dependencies@v2
        with:
          extra-packages: any::rcmdcheck, catmaid=natverse/rcatmaid
          needs: check
```

and the additional failure was an unrelated issue.

**Comment by jefferis** [comment]
closing with apologies for noise.

**Comment by gaborcsardi** [comment]
No worries.
